### PR TITLE
Return angles in order of application and add more tests.

### PR DESCRIPTION
**Pull request by de-vri-es**
This PR changes the returned angles to be in the order that they should be applied. This is quite important: when you have ZYX "Euler" angles, they are in the order ZYX. They are never ordered XYZ to be applied as ZYX.

Additionally, this PR adds a bunch of tests to verify the computation against nalgebra using randomly (but deterministically) generated input.

This PR will merge into the `initial-implementation` branch.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -25,8 +33,3 @@ jobs:
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
           args: --workspace --all-targets --all-features
-      - name: Cleanup
-        uses: actions-rs/cargo@v1
-        with:
-          command: sweep
-          args: --maxsize 5120 -r
```
Comment (hgaiser) [inline .github/workflows/rust.yml:32]: Hmm odd, thought I had also done these changes..

**Review by hgaiser — COMMENTED** [review]
I don't agree with reordering the return values, depending on in which order they are applied.

I imagine the main use case will be something like this:

1. Receive a pose with axis-angle notation for rotation.
2. Convert this to Euler angle notation.
3. Construct a robot pose, which as far as I have seen them is always in the form of `[x, y, z, rx, ry, rz]` regardless of the Euler convention used.

So you would get something along these lines:

```
pose = receive_pose()
orientation = convert_axis_angle_to_euler_zyx(pose.orientation)
target_pose = [
    pose.translation[0],
    pose.translation[1],
    pose.translation[2],
    orientation[0],
    orientation[1],
    orientation[2]
]
move(target_pose)
```

What you are proposing would be:

```
pose = receive_pose()
orientation = convert_axis_angle_to_euler_zyx(pose.orientation)
target_pose = [
    pose.translation[0],
    pose.translation[1],
    pose.translation[2],
    orientation[2],
    orientation[1],
    orientation[0]
]
move(target_pose)
```

> They are never ordered XYZ to be applied as ZYX.

Can you elaborate more on this? As I said I've only seen `rx, ry, rz` ordering. As an example in [this](https://www.electrobit.ee/web/file_bank/Manuals/robotid/mitsubishi/cr800/CR800-functions-detailed_ENG.pdf) manual for Mitsubishi CR800:

image

(for whatever reason they call `rx, ry, rz` as `A, B, C`)

**Comment by de-vri-es** [comment]
I think they have a poor wording, but what they describe is XYZ, only they call it ZYX "extrinsic", because each rotation is applied in the word frame. But ZYX each time in world frame is the same as XYZ in local rotated frame.

To summerize: they write XYZ and they do XYZ.

But this becomes even more obvious when you use proper euler angles, like `XY'X''`. What order will you choose if you want to stick with XYZ?

**Comment by de-vri-es** [comment]
> image

Hmm, it looks like their manual has a pretty significant typo:
> Also, rotation is set to take place in a predetermined sequence, and the amount of rotation is calculated (controlled) first for a rotation on the **Z** axis, followed by one on the **Y** axis and one on the **Z** axis in the order shown.

ZYZ? Doesn't match with the rest of the story. Looks like Mitsubishi doesn't even know the order >.<

My guess is, they interpret ABC as XYZ but with extrinsic rotations, which we would call ZYX.

/edit: And I wouldn't be surprised if this is the reason why their own manual is wrong. Got to love Euler and Tait-Bryan angles, no?

**Comment by hgaiser** [comment]
Hmm that sounds likely and fits with how these angles are used. It's difficult to write quality documentation I guess.

**Review by hgaiser — APPROVED** [review]

**Pull request merged**